pkg/deployer/platform/detect/detect.go

```diff
@@ -64,14 +64,18 @@ func Platform(ctx context.Context) (platform.Platform, error) {
 	if err != nil {
 		return platform.Unknown, err
 	}
-	return PlatformFromLister(ctx, ocpCli.ConfigV1.ClusterVersions())
+	return PlatformFromClients(ctx, ocpCli.ConfigV1.ClusterVersions(), ocpCli.ConfigV1.Infrastructures())
 }
 
 type ClusterVersionsLister interface {
 	List(ctx context.Context, opts metav1.ListOptions) (*ocpconfigv1.ClusterVersionList, error)
 }
 
-func PlatformFromLister(ctx context.Context, cvLister ClusterVersionsLister) (platform.Platform, error) {
+type InfrastructuresGetter interface {
+	Get(ctx context.Context, name string, options metav1.GetOptions) (*ocpconfigv1.Infrastructure, error)
+}
+
+func PlatformFromClients(ctx context.Context, cvLister ClusterVersionsLister, infraGetter InfrastructuresGetter) (platform.Platform, error) {
```

Comment:
let's try not to break backward compatibility so suddenly. Let's keep the old function (which will never return Hypershift) and let's add this one. When you do so please add a comment on `PlatformFromLister` like "deprecated, use PlatformFromClients", so in the next-next release or later we can remove it

Reply:
Ack. what about the unit-tests? keep the old one?

Reply:
> Ack. what about the unit-tests? keep the old ones as well?

every public function should be unit-tested if possible, unless there are trivial wrappers kept for backward compat. So I'd say yes, let's keep (some) old unit tests